be/src/exprs/cast_expr_json.cpp

```diff
@@ -40,7 +42,14 @@ class CastColumnItemVisitor final : public ColumnVisitorAdapter<CastColumnItemVi
 
     static Status cast_datum_to_json(const ColumnPtr& col, int row, const std::string& name, vpack::Builder* builder) {
         CastColumnItemVisitor visitor(row, name, builder);
-        return col->accept(&visitor);
+        try {
+            return col->accept(&visitor);
+        } catch (const arangodb::velocypack::Exception& e) {
+            std::string data = col->debug_item(row);
+            LOG(WARNING) << "cast to json failed: error_code=" << e.errorCode() << ", error_msg=" << e.what()
+                         << ", data=" << data;
+            return Status::DataQualityError("cast to json failed: " + data);
```

Comment:
why not put the whole error msg in the returning Status

Reply:
it's too complicated for normal user.